cpp/cuproj/README.md

```diff
@@ -1,7 +1,43 @@
 # <div align="left"><img src="https://rapids.ai/assets/images/rapids_logo.png" width="90px"/>&nbsp;cuProj: GPU-Accelerated Coordinate Projection</div>
 
+## Resources
+- [cuProj User's Guide](https://docs.rapids.ai/api/cuspatial/cuproj/stable/): Python API reference and guides
+- [cuProj Developer Documentation](https://docs.rapids.ai/api/cuspatial/cuproj/stable/): Understand cuProj's architecture
+- [Getting Started](https://docs.rapids.ai/install): Instructions for installing cuSpatial/cuProj
```

Comment:
Let's wait until we write these docs...

Reply:
Ha yeah this isn't really fully ready for review (hence draft) and that was 100% copilot hallucination I left in since I was uncertain if it existed or not but wanted to have a starting point.

Reply:
I filled in a lot of this this weekend. Only the developer guide is a bit lacking.

Reply:
Leaving cuProj readme work in #1237, this will just be for the cuSpatial readme linking back to cuProj + any required update-version.sh tweaks that are still needed after cuProj PRs are merged.